solana/solana-ibc/programs/solana-ibc/src/lib.rs

```diff
@@ -513,6 +519,75 @@ pub mod solana_ibc {
         let chain = &mut ctx.accounts.chain;
         chain.update_chain_config(config_payload)
     }
+
+    /// Method which updates the connection delay of a particular connection
+    ///
+    /// Fails if the connection doesnt exist.
+    /// Can only be called by fee collector.
+    pub fn update_connection_delay_period(
+        ctx: Context<UpdateConnectionDelay>,
+        connection_id: u64,
+        delay_period_in_ns: u64,
+    ) -> Result<()> {
+        let storage = &mut ctx.accounts.storage;
+
+        // Panic if connection_id doenst exist
+        if storage.connections.len() >= connection_id as usize {
+            return Err(error!(error::Error::InvalidConnectionId));
+        }
+
+        let provable = crate::storage::get_provable_from(
+            &ctx.accounts.trie,
+            &ctx.accounts.fee_collector,
+        )?;
+        let chain = &mut ctx.accounts.chain;
+
+        // Before anything else, try generating a new guest block.  However, if
+        // that fails it’s not an error condition.  We do this at the beginning
+        // of any request.
+        chain.maybe_generate_block(&provable)?;
+
+        let accounts = crate::TransferAccounts {
+            sender: None,
+            receiver: None,
+            token_account: None,
+            token_mint: None,
+            escrow_account: None,
+            mint_authority: None,
+            token_program: None,
+            fee_collector: None,
+        };
+
+        let mut store =
```

Comment:
Uh? Why not storage::from_ctx!?

Reply:
well then i had to rename the signer as `sender`, though it really doesnt matter what i name.

Reply:
I think it’s better to rename it so we don’t have to repeat the same code.  If we *really* care about the name there might be way to make it an option in the macro, but we probably don’t care enough.